GitUI/CommitInfo/CommitInfo.cs

```diff
@@ -144,6 +144,23 @@ public CommitInfo()
             rtbxCommitMessage.Height = 1;
         }
 
+        /// <summary>
+        /// Clean up any resources being used.
+        /// </summary>
+        /// <param name="disposing">true if managed resources should be disposed; otherwise, false.</param>
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _asyncLoadCancellation.CancelCurrent();
+                ThreadHelper.JoinPendingOperations(); // those run with FileAndForget
```

Comment:
Can this be reworked somehow?
This awaits all FileAndForget(), not just those created by this form.

When trying to find a solution for #9440 and add cancellation token that are disposed in REvDiff, RevGrid Dispose(), they are not executed until this has finished. So the tokens are not cancelled.